src/model/document.ts

```diff
@@ -268,14 +280,21 @@ export class Document implements Feature {
   }
 
   /**
-   * Adds and indexes a feature to this documentled before resolve().
+   * Return all warnings on local features contained within the document as well
+   * as warnings on features detected by following through all imports. If the
+   * deep argument is set to false, only return warnings found on local
+   * features.
    */
-  _addFeature(feature: Feature) {
-    if (this._doneResolving) {
-      throw new Error('_addFeature can not be called after _resolve()');
+  getWarnings(deep?: boolean): Warning[] {
+    const warnings: Warning[] = [];
+    if (deep == null) {
+      deep = true;
```

Comment:
coming back to this, I'll change the default behavior to default false for now because that's what I would ultimately expect from the method. But we should figure out our overall position here asap